README.md

```diff
@@ -1,6 +1,126 @@
+
+For the time being, we'll accept only two kind of Kamelets:
+
+- **Sources**: Kamelets producing data that can be forwarded to any chosen destination. In the Camel jargon, a source can be used consumer-side.
+Kamelets belonging to this category **MUST** be marked with label: `camel.apache.org/kamelet.type=source`.
+- **Sinks**: Kamelets that accept data with a specific datashape and forward it to an external system. In the Camel jargon, a sink can be used producer-side.
+Kamelets belonging to this category **MUST** be marked with label: `camel.apache.org/kamelet.type=sink`.
+
+All Kamelets **MUST** provide a value for label `camel.apache.org/kamelet.type`.
+
+All Kamelets **MUST** declare an icon in the `camel.apache.org/kamelet.icon` annotation using the embedded URL `data:image` format. An icon annotation **CANNOT** contain a link to an external location. A Kamelet **SHOULD** use the specific `data:image/svg+xml;base64` format whenever it's possible.
+
+Kamelets that are logically related (e.g. all Kamelets that allow doing things with Twitter) **SHOULD** be linked together using the label `camel.apache.org/kamelet.group=<name of the group>` (e.g. `camel.apache.org/kamelet.group=Twitter`) to ease visualization in tooling.
+
+We provide an example of Kamelet to give more context to the following sections:
+
+```yaml
+apiVersion: camel.apache.org/v1alpha1
+kind: Kamelet
+metadata:
+  name: timer-source
+  annotations:
+    camel.apache.org/kamelet.icon: data:image/svg+xml;base64,PD94...
+  labels:
+    camel.apache.org/kamelet.type: source
+spec:
+  definition:
+    title: Timer Source
+    description: Produces periodic events with a custom payload
+    required:
+      - message
+    properties:
+      period:
+        title: Period
+        description: The interval between two events
+        type: integer
+        default: 1000
+      message:
+        title: Message
+        description: The message to generate
+        type: string
+        example: hello world
+  types:
+    out:
+      mediaType: text/plain
+  flow:
+    from:
+      uri: timer:tick
+      parameters:
+        period: "{{period}}"
+      steps:
+        - set-body:
+            constant: "{{message}}"
+        - to: kamelet:sink
+
+```
+
+
+### Flow Code
+
+The Camel route that defines the behavior of the Kamelet **MUST** be provided in YAML flow syntax.
+The Kamelet **MAY** declare additional supporting routes that **MUST** be written in YAML syntax (to provide better support in all Camel subprojects).
+
+The code of a "source" Kamelet must send data to the `kamelet:sink` special endpoint. The code of a "sink" Kamelet must consume data from the special endpoint `kamelet:source`.
+
+The Kamelet **CAN** declare dependencies on Camel components using the syntax `camel:<component-name>` (e.g. `camel:telegram`).
```

Comment:
For these CAN it would be good to have an example showing that, eg can we add another out of the box kamelet that does that.

Also the timer kamelet does not declare that it uses camel:timer